namer/core/src/main/scala/io/buoyant/namer/DelegateTree.scala

```diff
@@ -130,20 +130,20 @@ object DelegateTree {
         DelegateTree.Union(path, dentry, delegates: _*)
     }
 
-  def fromNameTree[T](names: NameTree[T]): DelegateTree[T] =
+  def fromNameTree[T](path: Path, names: NameTree[T]): DelegateTree[T] =
```

Comment:
This is a pretty unfortunate area of the code.  Any time that you're converting a NameTree to a DelegateTree, the path property is meaningless.  We could set it to Path.empty all the time but DelegateTrees are actually used to render the namerd json API (uggggh) and we need to set path to null to make sure it doesn't show up in the json output.  

What about changing the method signature to:
`def fromNameTree[T](names: NameTree[T], path: Path = Path.empty): DelegateTree[T]`

and most callers can leave path unspecified.  Only the json api handlers need to explicitly pass null in.